diff --git include/llvm/IR/ValueMap.h include/llvm/IR/ValueMap.h
index f2ea405f..a1ab2a50 100644
--- include/llvm/IR/ValueMap.h
+++ include/llvm/IR/ValueMap.h
@@ -104 +104 @@ public:
-  bool hasMD() const { return MDMap; }
+  bool hasMD() const { return bool(MDMap); }
diff --git projects/compiler-rt/lib/sanitizer_common/sanitizer_linux.cc projects/compiler-rt/lib/sanitizer_common/sanitizer_linux.cc
index 0e776b90..5088f2d0 100644
--- projects/compiler-rt/lib/sanitizer_common/sanitizer_linux.cc
+++ projects/compiler-rt/lib/sanitizer_common/sanitizer_linux.cc
@@ -588,2 +588 @@ uptr internal_prctl(int option, uptr arg2, uptr arg3, uptr arg4, uptr arg5) {
-uptr internal_sigaltstack(const struct sigaltstack *ss,
-                         struct sigaltstack *oss) {
+uptr internal_sigaltstack(const void *ss, void *oss) {
diff --git projects/compiler-rt/lib/sanitizer_common/sanitizer_linux.h projects/compiler-rt/lib/sanitizer_common/sanitizer_linux.h
index 3013c25f..d7d4db89 100644
--- projects/compiler-rt/lib/sanitizer_common/sanitizer_linux.h
+++ projects/compiler-rt/lib/sanitizer_common/sanitizer_linux.h
@@ -23 +22,0 @@ struct link_map;  // Opaque type returned by dlopen().
-struct sigaltstack;
@@ -32,2 +31 @@ uptr internal_getdents(fd_t fd, struct linux_dirent *dirp, unsigned int count);
-uptr internal_sigaltstack(const struct sigaltstack* ss,
-                          struct sigaltstack* oss);
+uptr internal_sigaltstack(const void* ss, void* oss);
diff --git projects/compiler-rt/lib/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc projects/compiler-rt/lib/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc
index d20b5248..33894054 100644
--- projects/compiler-rt/lib/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc
+++ projects/compiler-rt/lib/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cc
@@ -239 +239 @@ static int TracerThread(void* argument) {
-  struct sigaltstack handler_stack;
+  stack_t handler_stack;
diff --git projects/compiler-rt/lib/tsan/rtl/tsan_platform_linux.cc projects/compiler-rt/lib/tsan/rtl/tsan_platform_linux.cc
index 4dcfa558..8ce75be1 100644
--- projects/compiler-rt/lib/tsan/rtl/tsan_platform_linux.cc
+++ projects/compiler-rt/lib/tsan/rtl/tsan_platform_linux.cc
@@ -358 +358 @@ int ExtractResolvFDs(void *state, int *fds, int nfd) {
-  __res_state *statp = (__res_state*)state;
+  struct __res_state *statp = (struct __res_state*)state;
