*** include/llvm/ADT/IntrusiveRefCntPtr.h       2014-07-06 00:20:59.000000000 +0200
--- include/llvm/ADT/IntrusiveRefCntPtr_new.h   2016-06-21 10:04:15.610143165 +0200
***************
*** 154,159 ****
--- 154,162 ----
      }

      template <class X>
+     friend class IntrusiveRefCntPtr;
+
+     template <class X>
      IntrusiveRefCntPtr(IntrusiveRefCntPtr<X>&& S) : Obj(S.get()) {
        S.Obj = 0;
      }
