diff --git a/src/parser/smtlib2.y b/src/parser/smtlib2.y
index a94bd6c..5263bf5 100644
--- a/src/parser/smtlib2.y
+++ b/src/parser/smtlib2.y
@@ -64,6 +64,7 @@
     FatalError("");
     return 1;
   }
+  int yyerror(void* AssertsQuery, const char* s) { return yyerror(s); }
 
   ASTNode querysmt2;
   ASTVec assertionsSMT2;
@@ -72,9 +73,10 @@
 #define YYMAXDEPTH 104857600
 #define YYERROR_VERBOSE 1
 #define YY_EXIT_FAILURE -1
-#define YYPARSE_PARAM AssertsQuery
   %}
 
+%parse-param {void* AssertsQuery}
+
 %union {  
   unsigned uintval;                  /* for numerals in types. */
   //ASTNode,ASTVec
